jobs/resource_job.go

```diff
@@ -728,6 +736,8 @@ var jobSchema = common.StructToSchema(JobSettings{},
 		s["continuous"].ConflictsWith = []string{"schedule", "trigger"}
 		s["trigger"].ConflictsWith = []string{"schedule", "continuous"}
 
+		s["trigger"].ExactlyOneOf = []string{"file_arrival", "table_update"}
```

Comment:
because it is a nested field, you need to follow the below example
```		
// we need to have only one of user name vs service principal in the run_as block
run_as_eoo := []string{"run_as.0.user_name", "run_as.0.service_principal_name"}
common.MustSchemaPath(s, "run_as", "user_name").ExactlyOneOf = run_as_eoo
common.MustSchemaPath(s, "run_as", "service_principal_name").ExactlyOneOf = run_as_eoo
```